### PR TITLE
Replace RPC with Scheduler HTTP API

### PR DESCRIPTION
**Pull request by Matt711**
This PR replaces the call to the RPC object in favor of the new scheduler HTTP API. 
Related Issue: [#5935](https://github.com/dask/distributed/issues/5935)
Related PR: [#6270](https://github.com/dask/distributed/pull/6270)

### CODE REVIEW

**Review by jacobtomlinson — COMMENTED** [review]
Nice to see this! A couple of comments.

Also, this does depend on an unreleased feature in `distributed` so we need to wait for a release before we can merge this and bump the minimum required distributed version.

File: dask_kubernetes/operator/operator.py
Comment (Matt711) [reply dask_kubernetes/operator/operator.py:225]: They should be deleted in the next few lines when `api.delete_namespaced_pod` is called.

Comment (Matt711) [reply dask_kubernetes/operator/operator.py:225]: They should be deleted in the next few lines when `api.delete_namespaced_pod` is called.

Comment (jacobtomlinson) [reply dask_kubernetes/operator/operator.py:225]: Yup my bad! GitHub collapsed that and I missed it.

File: dask_kubernetes/operator/operator.py
Comment (jacobtomlinson) [inline dask_kubernetes/operator/operator.py:204]: This spec could be very verbose. I'd prefer not to log it.

File: dask_kubernetes/operator/operator.py
Comment (jacobtomlinson) [inline dask_kubernetes/operator/operator.py:12]: We probably still need to do this as this handles accessing the scheduler in a variety of different service modes. Although we don't want the comm anymore we want the HTTP server (port `8787`). 

Luckily @mmccarty has taken us part of the way down this path by making the port forwarding code more configurable in #489.

We probably still need to expand `get_scheduler_address` and `get_external_address_for_scheduler_service` in `dask_kubernetes.common.networking` to also accept a kwarg to choose which port we want to forward. Right now we assume the port is called `"comm"` but this should be configurable as we would need to set it to `"dashboard"` here.

https://github.com/dask/dask-kubernetes/blob/feedbf40b974774c53f73e02cb3ea3fd72192c14/dask_kubernetes/common/networking.py#L20

File: dask_kubernetes/operator/operator.py
Comment (jacobtomlinson) [inline dask_kubernetes/operator/operator.py:217]: We can't assume this will be on `localhost`. We still need the `get_scheduler_service` call, but we need to modify that function so we can tell it we want port `8787` rather than `8786`. See my other comment about this.

**Review by jacobtomlinson — APPROVED** [review]
Thanks for getting these fixes in @Matt711.

I tested this locally and it didn't quite seem to work so I pushed another commit with a fix.

This should be good to merge once dask/community/issues/245 is done and we can set `distributed>=2022.05.1` in `requirements.txt`.

**Comment by mmccarty** [comment]
> Also, this does depend on an unreleased feature in distributed so we need to wait for a release before we can merge this and bump the minimum required distributed version.

Did we get an answer to this?

**Comment by jacobtomlinson** [comment]
@mmccarty see https://github.com/dask/dask-kubernetes/pull/499#pullrequestreview-979853631, just waiting on the next release now before we can merge 😁

**Comment by jacobtomlinson** [comment]
Given that the HTTP API is still disabled by default in the latest release I've implemented some fallback options:
- If the operator fails to use the HTTP API it will try the RPC
- If the operator fails to use the RPC it will fall back to dumb last-in-first-out scaling down

This should unblock us to merge now and also make it easier to support multiple Dask versions in the future.

**Pull request merged**
